ftw/upgrade/testing.py

```diff
@@ -52,6 +52,17 @@ def setUpZope(self, app, configurationContext):
 
         z2.installProduct(app, 'Products.CMFPlacefulWorkflow')
 
+        try:
+            # Plone 4 with collective.indexing
+            pkg_resources.get_distribution('collective.indexing')
+
+            import collective.indexing
+            xmlconfig.file('configure.zcml', collective.indexing,
+                           context=configurationContext)
+            z2.installProduct(app, 'collective.indexing')
+        except pkg_resources.DistributionNotFound:
```

Comment:
Since we expect the `DistributionNotFound` to be happing in the first line, I would move the other lines out of the `try` block.
Theoretically we could be able to import `collective.indexing` but within it a `DistributionNotFound` is raised - we would catch that by accident here.

I usually do that like this:
```python
        try:
            # Plone 4 with collective.indexing
            pkg_resources.get_distribution('collective.indexing')
        except pkg_resources.DistributionNotFound:
            pass
        else:
            import collective.indexing
            xmlconfig.file('configure.zcml', collective.indexing,
                           context=configurationContext)
            z2.installProduct(app, 'collective.indexing')
```